src/protocol/parser.rs

```diff
@@ -0,0 +1,130 @@
+use crate::R;
```

Comment:
I would call this module `argument_parser`. Or something that distinguishes this from the yaml parsing in `protocol`.